src/Compilers/CSharp/Test/Emit3/RefStructInterfacesTests.cs

```diff
@@ -23232,5 +23218,669 @@ .maxstack  1
+@"[Test]: Return type is ByRef, TypedReference, ArgHandle, or ArgIterator. { Offset = 0x20 }
+[get_PT]: Call not allowed on abstract methods. { Offset = 0x16 }
+[get_PT]: Missing callvirt following constrained prefix. { Offset = 0x16 }
+[Create]: Return type is ByRef, TypedReference, ArgHandle, or ArgIterator. { Offset = 0x9 }") :
+                    Verification.Skipped).VerifyDiagnostics();
+        }
+
+        [Fact]
+        public void ObjectCreation_04()
+        {
+            var text = @"
+class Program
+{
+    static C<T> Test<T>()
+        where T : struct, I1, allows ref struct
+    {
+        return new C<T>() { PT = { P = 123 } };
+    }
+
+    static void Main()
+    {
+        System.Console.Write(Test<S>().PT.P);
+    }
+}
+
+ref struct C<T>
+    where T : struct, I1, allows ref struct
+{
+    public T PT {get;set;}
+}
+
+public ref struct S : I1
+{
+    public int P {get;set;}
+}
+
+public interface I1
+{
+    public int P {get;set;}
+}
+";
+
+            var comp = CreateCompilation(text, targetFramework: s_targetFrameworkSupportingByRefLikeGenerics, options: TestOptions.ReleaseExe);
+
+            comp.VerifyDiagnostics(
+                // (7,29): error CS1918: Members of property 'C<T>.PT' of type 'T' cannot be assigned with an object initializer because it is of a value type
+                //         return new C<T>() { PT = { P = 123 } };
+                Diagnostic(ErrorCode.ERR_ValueTypePropertyInObjectInitializer, "PT").WithArguments("C<T>.PT", "T").WithLocation(7, 29)
+                );
+        }
+
+        [Theory]
+        [CombinatorialData]
+        public void ObjectCreation_05(bool addStructConstraint)
+        {
+            var text = @"
+class Program
+{
+    static T Test<T>()
+        where T : " + (addStructConstraint ? "struct, I1" : "I1, new()") + @", allows ref struct
+    {
+        return new T() { 100, 20, 3 };
+    }
+
+    static void Main()
+    {
+        System.Console.Write(Test<S>().P);
+        System.Console.Write("" "");
+        System.Console.Write((new S() { 200, 40, 6 }).P);
+    }
+}
+
+public ref struct S : I1
+{
+    public int P {get;set;}
+    public void Add(int x) => P += x;
+    System.Collections.IEnumerator System.Collections.IEnumerable.GetEnumerator() => null;
+}
+
+public interface I1 : System.Collections.IEnumerable
+{
+    public void Add(int x);
+}
+
+namespace System
+{
+    public class Activator
+    {
+        public static T CreateInstance<T>() where T : allows ref struct => default;
+    }
+}
+";
+
+            var comp = CreateCompilation(text, targetFramework: s_targetFrameworkSupportingByRefLikeGenerics, options: TestOptions.ReleaseExe);
+
+            var verifier = CompileAndVerify(
+                comp,
+                expectedOutput: ExecutionConditionUtil.IsMonoOrCoreClr ? "123 246" : null,
+                verify: ExecutionConditionUtil.IsMonoOrCoreClr ? Verification.Passes : Verification.Skipped).VerifyDiagnostics();
+        }
```

Comment:
>} [](http://example.com/codeflow?start=8&length=1)

Are we testing the corresponding cases for collection expressions?

Reply:
> Are we testing the corresponding cases for collection expressions?

`CollectionExpressions_01`?